src/Storages/IStorageCluster.cpp

```diff
@@ -73,13 +78,21 @@ void IStorageCluster::read(
     SelectQueryInfo & query_info,
     ContextPtr context,
     QueryProcessingStage::Enum processed_stage,
-    size_t /*max_block_size*/,
-    size_t /*num_streams*/)
+    size_t max_block_size,
+    size_t num_streams)
 {
+    auto cluster_name_ = getClusterName(context);
```

Comment:
Why the underscore in the variable name?

Reply:
Ah, I assume it's because `IStorageCluster` already has a `cluster_name` member that comes from engines like `S3Cluster` and not the setting, right?

If that's so, what happens if the user has both of them set?

PLus, I would rename it to be something like `cluster_name_from_settings`

Reply:
Warning that local variable masked class member, and option "warning as error" (`-Werror`).